### PR TITLE
2705 upgrade island python version

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Issue #2705

Upgrades the Island Python version to 3.11.2

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running ETE tests
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: build_scripts/docker/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,27 @@
+ARG PYTHON_VERSION=3.11
 # Install python dependencies using the bitnami/python:3.7 image, which includes
 # development dependencies.
```
Comment (shreyamalviya) [inline build_scripts/docker/Dockerfile:3]: Update comment

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Some of the initial commits' messages need to be prepended with "Build: ".

**Review by VakarisZ — COMMENTED** [review]
We'll have to update `docs/content/reference/operating_systems_support.md:47` Not sure in which PR though, but it's a small change

**Comment by mssalvatore** [comment]
> We'll have to update `docs/content/reference/operating_systems_support.md:47` Not sure in which PR though, but it's a small change

This was done in https://github.com/guardicore/monkey/pull/3015. What did we miss?

**Comment by VakarisZ** [comment]
> @VakarisZ This was done in https://github.com/guardicore/monkey/pull/3015. What did we miss?
Didn't notice. Ignore my comment then

**Review by shreyamalviya — APPROVED** [review]
Approved but fix the initial commit messages before merging

**Pull request merged**